hedera-node/hedera-app/src/main/java/com/hedera/node/app/records/impl/BlockRecordManagerImpl.java

```diff
@@ -346,6 +364,26 @@ public void advanceConsensusClock(@NonNull final Instant consensusTime, @NonNull
         this.lastBlockInfo = newBlockInfo;
     }
 
+    /**
+     * Check if the consensus time of the last handled transaction is the default value. This is
+     * used to determine if migration records should be streamed
+     *
+     * @param blockInfo the block info object to test
+     * @return true if the given block info has a last handled transaction time that is considered a
+     * 'default' or 'unset' value, false otherwise.
+     */
+    public static boolean isDefaultConsTimeOfLastHandledTxn(@Nullable final BlockInfo blockInfo) {
+        if (blockInfo == null || blockInfo.consTimeOfLastHandledTxn() == null) {
+            return true;
+        }
```

Comment:
same here I think @mhess-swl

Reply:
Again, only a defensive check. This method will still return the correct value in the `EPOCH` case, and there's a corresponding test case to validate this.